src/systemTest/java/soot/defaultInterfaceMethods/DefaultInterfaceTest.java

```diff
@@ -0,0 +1,897 @@
+    SootMethod refDefaultRead = resolveMethodRefInBody(mainBody.getUnits(), "void read()");
+    SootMethod refDefaultWrite = resolveMethodRefInBody(mainBody.getUnits(), "void write()");
+
+    SootMethod resolvedMainMethod =
+        VirtualCalls.v()
+            .resolveNonSpecial(
+                Scene.v().getRefType(testClassSig), mainPrintMethod.makeRef(), false);
+    SootMethod resolvedWritePrintMethod =
+        VirtualCalls.v()
+            .resolveNonSpecial(
+                Scene.v().getRefType(testClassSig), writeInterfacePrint.makeRef(), false);
+    SootMethod resolvedReadPrintMethod =
+        VirtualCalls.v()
+            .resolveNonSpecial(
+                Scene.v().getRefType(testClassSig), readInterfacePrint.makeRef(), false);
+    SootMethod resolvedDefaultReadMethod =
+        VirtualCalls.v()
+            .resolveNonSpecial(
+                Scene.v().getRefType(testClassSig), readInterfaceRead.makeRef(), false);
+    SootMethod resolvedDefaultWriteMethod =
+        VirtualCalls.v()
+            .resolveNonSpecial(
+                Scene.v().getRefType(testClassSig), writeInterfaceWrite.makeRef(), false);
+
+    FastHierarchy fh = Scene.v().getFastHierarchy();
+    SootMethod concreteImplMainPrint = fh.resolveConcreteDispatch(testClass, mainPrintMethod);
+    SootMethod concreteImplWritePrint = fh.resolveConcreteDispatch(testClass, refWritePrintMethod);
+    SootMethod concreteImplReadPrint = fh.resolveConcreteDispatch(testClass, refReadPrintMethod);
+    SootMethod concreteImplDefaultRead = fh.resolveConcreteDispatch(testClass, refDefaultRead);
+    SootMethod concreteImplDefaultWrite = fh.resolveConcreteDispatch(testClass, refDefaultWrite);
+
+    assertEquals(
+        Sets.newHashSet(readInterfaceRead),
+        fh.resolveAbstractDispatch(interfaceRead, refDefaultRead));
+    assertEquals(
+        Sets.newHashSet(writeInterfaceWrite),
+        fh.resolveAbstractDispatch(interfaceWrite, refDefaultWrite));
+    assertEquals(
+        Sets.newHashSet(mainPrintMethod),
+        fh.resolveAbstractDispatch(interfaceRead, refReadPrintMethod));
+    assertEquals(
+        Sets.newHashSet(mainPrintMethod),
+        fh.resolveAbstractDispatch(interfaceWrite, refWritePrintMethod));
+
+    /* Edges should be present */
+    boolean edgeMainPrintToReadPrint = checkInEdges(readInterfacePrint, mainPrintMethod);
+    boolean edgeMainPrintToWritePrint = checkInEdges(writeInterfacePrint, mainPrintMethod);
+    boolean edgeMainMethodToPrint = checkInEdges(mainPrintMethod, target);
+    boolean edgeMainMethodToReadMethod = checkInEdges(readInterfaceRead, target);
+    boolean edgeMainMethodToWriteMethod = checkInEdges(writeInterfaceWrite, target);
+
+    /* Edges should not be present */
+    boolean edgeMainMethodToReadPrint = checkInEdges(readInterfacePrint, target);
+    boolean edgeMainMethodToWritePrint = checkInEdges(writeInterfacePrint, target);
+
+    final ReachableMethods reachableMethods = Scene.v().getReachableMethods();
```

Comment:
Could you please add your test case `CallGraphTest` here as separet @Test method? Move the target files to the `soot.defaultInterfaceMethods` directory in test-targets and then copy paste the head of one of this test methods with the correct target class: `  String testClassSig = "soot.defaultInterfaceMethods.PavansSuperCoolTest";`

Reply:
Okay